core/src/main/scala/caliban/GraphQLAspect.scala

```diff
@@ -16,3 +19,19 @@ trait GraphQLAspect[+LowerR, -UpperR] { self =>
         other(self(gql))
     }
 }
+
+object GraphQLAspect {
+
+  def withSchemaDirectives(directives: List[Directive]): GraphQLAspect[Nothing, Any] = new GraphQLAspect[Nothing, Any] {
+    def apply[R](gql: GraphQL[R]): GraphQL[R] = gql.withSchemaDirectives(directives)
+  }
+
+  def withDirectives(directives: List[__Directive]): GraphQLAspect[Nothing, Any] = new GraphQLAspect[Nothing, Any] {
+    def apply[R](gql: GraphQL[R]): GraphQL[R] = gql.withAdditionalDirectives(directives)
+  }
+
+  def withTypes(types: List[__Type]): GraphQLAspect[Nothing, Any] = new GraphQLAspect[Nothing, Any] {
+    def apply[R](gql: GraphQL[R]): GraphQL[R] = gql.withAdditionalTypes(types)
```

Comment:
I didn't end up needing these, but I think they are useful so I kept them in place. It makes it easy if you are defining a new aspect that also requires new directives/types.